cluster/images/etcd/migrate/migrate_server.go

```diff
@@ -103,7 +103,7 @@ func (r *EtcdMigrateServer) Stop() error {
 		return fmt.Errorf("error sending SIGINT to etcd for graceful shutdown: %v", err)
 	}
 	gracefulWait := time.Minute * 2
-	stopped := make(chan bool)
+	stopped := make(chan bool, 1)
```

Comment:
In this case, switching to time.After will let you delete a bunch of poorly thought-out code :)

P.S. a better fix is to close the channel instead of trying to write to it.

Reply:
+1 to using time.After (and cleaning up this code I wrote!)

Reply:
@lavalamp thanks. Changes done.

Reply:
@lavalamp @jparklab sorry, I missed this part, you mean replace time.Sleep() with time.After() in the goroutine ?

Reply:
No, delete the channel and the goroutine completely. Replace by a call to time.After in the select statement.

Reply:
Changes done. Thank you.